src/lib.rs

```diff
@@ -2372,12 +2371,9 @@ pub mod config {
             }
         }
         /// Parse with context.
-        fn do_parse<R: io::Read>(
-            &mut self,
-            context: &mut HtmlContext,
-            input: R,
-        ) -> Result<RenderTree> {
-            super::parse_with_context(input, context)
+        pub(crate) fn do_parse<R: io::Read>(&self, input: R) -> Result<RenderTree> {
```

Comment:
I'm not convinced about this change.
It's true that currently there's no state kept across the stages, but I think it's worthwhile piping the same context through so that there's somewhere to add such information for future features/configuration.
As a possible example, we might want to keep the CSS style data around to support media queries and toggle between light/dark mode without re-parsing every time.